tools/lsp/common.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use i_slint_compiler::CompilerConfiguration;
+use std::path::{Path, PathBuf};
+
+use crate::ServerNotifier;
+
+// Common data structures:
+
+/// API used by the LSP to talk to the Preview.
+pub struct PreviewApi {
```

Comment:
Why is this not a trait?

Also please add comment which function are meant to be called by the preview, and which are meant to be called by the LSP